### PR TITLE
Login/Signup Pages

### PR DESCRIPTION
**Pull request by hodge-jai**
Note: The large diff mostly comes from changes to package-lock.json that occurred when running npm install.

### What this PR does (required):
- brings login and sign up pages in line with the mock ups

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

Screen Shot 2021-11-08 at 6 58 10 PM

Screen Shot 2021-11-08 at 6 57 38 PM

### Any information needed to test this feature (required):
- Need to test the responsiveness of the page for different screen sizes

### CODE REVIEW

File: client/src/components/AuthHeader/useStyles.ts
Comment (rajivtitus) [inline client/src/components/AuthHeader/useStyles.ts]: I know this is a part of the starter code but most of these styles can be applied to the Box component directly (it is a good practice to pass props to MUI components unless we explicitly need a class to apply some styles) 😄 

Eg: `<Box display="flex" alignItems="center" justifyContent="center"...>`

File: client/src/components/NavBar/NavBar.tsx
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: You can combine the MUI button with react-router's Link :)

Eg: 
```
<Button component={Link} to="/about" variant="contained" color="primary">
  About Page
</Button>
```

File: client/src/components/NavBar/NavBar.tsx
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: Ditto as comment above!

File: client/src/components/NavBar/NavBar.tsx
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: Please refrain from using CAPITALCASE text as some screen readers interpret them as abbreviations, you can use a CSS class to transform text instead.

File: client/src/components/NavBar/NavBar.tsx
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:10]: The NavBar is a separate feature, good work completing it within this PR. However, in the future please make sure you create a new PR for each feature :)

File: client/src/components/NavBar/useStyles.ts
Comment (rajivtitus) [inline client/src/components/NavBar/useStyles.ts]:
```suggestion
  loginSignup: {
```
Nitpick: Please make sure you use camelCase when writing CSS in JS 😄

File: client/src/themes/theme.ts
Comment (rajivtitus) [inline client/src/themes/theme.ts:5]: Nice work updating the styles within the theme 💯

File: client/src/pages/SignUp/SignUpForm/useStyles.ts
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/useStyles.ts]: You can add these colours to a palette within MUI's theme and then use the theme to apply them to ensure consistency across the app!

File: client/src/pages/SignUp/useStyles.ts
Comment (rajivtitus) [inline client/src/pages/SignUp/useStyles.ts]: Please avoid setting the font family outside of MUI's theme, you can use the theme to apply it instead!

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: Optional: Maybe we can rename this component? Calling it AuthHeader doesn't seem appropriate 😄

File: client/src/components/AuthHeader/useStyles.ts
Comment (hodge-jai) [reply client/src/components/AuthHeader/useStyles.ts]: Changes have been made. Can I go ahead and merge this?

Comment (rajivtitus) [reply client/src/components/AuthHeader/useStyles.ts]: Please wait for approval from a Hatchways team member prior to merging PRs :)

**Comment by rajivtitus** [comment]
@hodge-jai Please address the pending comments and tag me in for another review once you are done :)

**Comment by hodge-jai** [comment]
@rajivtitus Changes have been made. Please review again!

**Review by rajivtitus — COMMENTED** [review]
Reposting the additional comments. Please have a look and address :)

**Comment by hodge-jai** [comment]
@rajivtitus Please take a look at my changes and review!

**Review by rajivtitus — APPROVED** [review]
Excellent work making those changes! 💯

**Pull request merged**
